cli: harmonize click styled echos

Sinclert opened the pull request:
This PR harmonizes the utilization of `click.secho(...)` and `click.echo(click.style(...))`, to the former.

Throughout the codebase, there are many calls to [click.secho](https://click.palletsprojects.com/en/8.0.x/api/#click.secho), which provides an _styled_ echo functionality. This would be totally equivalent to calling `click.style` within a `click.echo` block, so any occurrences of the their concatenation have been replaced by [click.secho](https://click.palletsprojects.com/en/8.0.x/api/#click.secho) calls.

mvidalgarcia reviewed (commented):
Hi @Sinclert and thanks for the contribution! The changes you've made already help a lot and it's an important step to harmonize the r-client output messages. We'd like to go further harmonizing and make use of our [`display_message`](https://github.com/reanahub/reana-client/blob/4cd2d54137e50c10c809104fbed74de6ceedda64/reana_client/printer.py#L21) function to have everything centralized. Do you think you can make use of this function? It's also useful to have the same `==> ERROR/WARNING/...` prefixes everywhere to have a more consistent CLI.

PS. Needs rebase.

Sinclert commented:
You are welcome! I try to contribute back every-time I see something off on the codebases 😄 

Regarding the PR, I just rebased my branch, and changed the substituted function to the one you commented ([`display_message`](https://github.com/reanahub/reana-client/blob/4cd2d54137e50c10c809104fbed74de6ceedda64/reana_client/printer.py#L21) function). This time, in addition to substituting every `click.echo` and `click.secho` I could find, I also:

- Removed every `==> ...` message prefix, as the display_message function does it automatically (https://github.com/reanahub/reana-client/pull/553/commits/b15ff394f3751be07a0c5d8490cca5748a33ad56).
- Replaced `click.secho` target logic deciding the color of each message (https://github.com/reanahub/reana-client/pull/553/commits/c5b43c9f83ad1f08ad3f5e18167628bb6e10d53a).

On the second point, there is a [particular snippet](https://github.com/reanahub/reana-client/blob/3814b05465dc79ad682a14c84e3ae22db1b4eea8/reana_client/cli/quotas.py#L114) which depends on reana_commons. As I assumed that the referenced table of _health-status_ to _message-color_ is used in other components, I substituted it a hardcoded dictionary:

https://github.com/reanahub/reana-client/blob/c5b43c9f83ad1f08ad3f5e18167628bb6e10d53a/reana_client/cli/quotas.py#L111-L115

Sinclert commented:
Finally, let me know if you want to make an exception with the [`version` command](https://github.com/reanahub/reana-client/blob/c5b43c9f83ad1f08ad3f5e18167628bb6e10d53a/reana_client/cli/ping.py#L82). I know the [`display_message`](https://github.com/reanahub/reana-client/blob/4cd2d54137e50c10c809104fbed74de6ceedda64/reana_client/printer.py#L21) function introduces some padding that it might not be useful in this case.

VMois commented on reana_client/cli/utils.py:
```diff
@@ -315,10 +294,10 @@ def output_user_friendly_logs(workflow_logs, steps):
-        "running": "bright_blue",
-        "finished": "green",
-        "failed": "red",
+    status_to_msg_type = {
+        "running": "info",
+        "finished": "success",
+        "failed": "error",
```
*suggestion*: move it to constant dict to `config.py` as `JOB_STATUS_TO_MSG_TYPE`

Sinclert replied:
Done in https://github.com/reanahub/reana-client/pull/553/commits/5958a7db472f5a71d7ac48e39ea65d8c90af1873

VMois commented on reana_client/utils.py:
```diff
@@ -410,8 +388,8 @@ def run_command(cmd, display=True, return_output=False, stderr_output=False):
-        click.secho("[{0}] ".format(now), bold=True, nl=False, fg="green")
-        click.secho("{0}".format(cmd), bold=True)
+        display_message("[{0}] ".format(now), msg_type="success")
+        display_message("{0}".format(cmd))
```
*thought*: not sure how important is `bold` in the output. For me, it doesn't make a difference but leaving this comment here just in case.

Sinclert replied:
The `display_message` function cannot receive a `bold` flag. We could expose that argument, but I think changing the `display_message` function is not the scope of this PR.

I assumed is not relevant.

VMois commented on reana_client/cli/workflow.py:
```diff
@@ -1069,22 +1050,22 @@ def workflow_run(
     ctx.invoked_by_subcommand = True
     ctx.workflow_name = ""
-    click.secho("[INFO] Creating a workflow...", bold=True)
+    display_message("[INFO] Creating a workflow...")
```
*suggestion*: msg_type="info" instead of "[INFO]"

Sinclert replied:
Done in https://github.com/reanahub/reana-client/pull/553/commits/b04b1d606068ae6ca9fb898509f26fcabf43b1cd

VMois commented on reana_client/cli/workflow.py:
```diff
@@ -1069,22 +1050,22 @@ def workflow_run(
         access_token=access_token,
     )
-    click.secho("[INFO] Uploading files...", bold=True)
+    display_message("[INFO] Uploading files...")
```
The same as above

Sinclert replied:
Done in https://github.com/reanahub/reana-client/pull/553/commits/b04b1d606068ae6ca9fb898509f26fcabf43b1cd

VMois commented on reana_client/cli/workflow.py:
```diff
@@ -1069,22 +1050,22 @@ def workflow_run(
         access_token=access_token,
     )
-    click.secho("[INFO] Starting workflow...", bold=True)
+    display_message("[INFO] Starting workflow...")
```
The same as above

Sinclert replied:
Done in https://github.com/reanahub/reana-client/pull/553/commits/b04b1d606068ae6ca9fb898509f26fcabf43b1cd

VMois commented on reana_client/cli/workflow.py:
```diff
@@ -1204,16 +1183,15 @@ def workflow_diff(
-                line_color = "green"
-            click.secho(line, fg=line_color)
+                msg_type = "success"
+            display_message(line, msg_type=msg_type)
```
*suggestion*: conceptually it is not really good that differences in lines are marked with "error", "success", etc. instead of colors. I would leave colors here as an exception. In addition, msg_types can introduce unwanted formatting and formatting is very important for checking differences.

Sinclert replied:
Not sure what you mean by _conceptually_.

In any case, if you would like to use _colors_ instead of the high-level _msg_type_ argument, the `display_message` function would need to be replaced by `click.secho` here (which defeats the purpose of centralising everything, but we could make an exception.).

Want to leave `click.secho`?

VMois replied:
Let me check how `display_message` will affect lines formatting. I will post here the findings.

VMois replied:
What I meant by "conceptually": the difference is showing that some lines were removed (`-`), this is not an `error` or `success`, this is just stating "this line is removed", but the code that prints those removed lines is saying`msg_type="error"` which states this is an error.

VMois replied:
@Sinclert let's keep the original version of difference cmd. As you can see from screenshots that @mvidalgarcia posted a new version looks overwhelming. Thank you.

Sinclert replied:
Reverted in https://github.com/reanahub/reana-client/pull/553/commits/cbe2b47c260ae496cfbddedb33233dc5f79d770d

VMois commented on reana_client/cli/workflow.py:
```diff
@@ -1310,27 +1277,26 @@ def workflow_open_interactive_session(
             )
-            click.echo(
+            display_message(
                 "It could take several minutes to start the " "interactive session."
```
```suggestion
                "It could take several minutes to start the interactive session."
```

VMois reviewed (changes requested):
@Sinclert Could you, please, go through those 7 files and remove unused imports? I saw a couple in multiple places, would be good to keep it clean.

Otherwise, I left some suggestions, please, take a look at them.

Thank you very much for your work.

Sinclert commented on reana_client/cli/quotas.py:
```diff
@@ -106,22 +103,22 @@ def quota_show(
+                    "healthy": "success",
+                    "warning": "warning",
+                    "critical": "error",
+                }.get(health)
```
Done in https://github.com/reanahub/reana-client/pull/553/commits/638719fae1187c8e24b323bfcbee326335cd370f

Sinclert commented on reana_client/cli/ping.py:
```diff
@@ -39,23 +40,21 @@ def ping(ctx, access_token):  # noqa: D301
+                response.get("email"),
+                response.get("status"),
+            ),
+            msg_type=msg_type,
```
Done in https://github.com/reanahub/reana-client/pull/553/commits/b9e4d57dae5adc9075701bfcd9c2afd6d9f21983

Sinclert commented on reana_client/cli/ping.py:
```diff
@@ -80,4 +79,4 @@ def version(ctx):  # noqa: D301
     \t $ reana-client version
     """
-    click.echo(__version__)
+    display_message(__version__)
```
Ok. Commented just to be mindful that the `display_message` may change in the future.

Sinclert commented:
Hi @VMois 

Thanks for your review.

I just committed most of your suggestions, except from the one made on [this comment](https://github.com/reanahub/reana-client/pull/553#discussion_r704471000) (please review). Finally, I got rid of the unused imports that you mentioned, on commit https://github.com/reanahub/reana-client/pull/553/commits/ee97d2c2b2e906673dac0e0543a96796a28db83e

mvidalgarcia reviewed (commented):
I'm comparing this PR to the current master in two different virtual environments and there are some differences:

- `ping`
master:
![image](https://user-images.githubusercontent.com/8863238/132822262-2a40a6b2-d473-4211-9ec9-fca45981c248.png)
PR:
![image](https://user-images.githubusercontent.com/8863238/132822291-373ee6e0-59e0-4312-b30e-4f4b04237527.png)

- `list`
master:
![image](https://user-images.githubusercontent.com/8863238/132822552-7c64f6da-26a2-4668-9dfc-0d3055afacec.png)
PR:
![image](https://user-images.githubusercontent.com/8863238/132822583-901a9072-2920-4380-a81d-29bfe543d635.png)
- `diff`
master:
![image](https://user-images.githubusercontent.com/8863238/132823312-1d251766-5138-494f-b035-7ee3e551703f.png)
PR:
![image](https://user-images.githubusercontent.com/8863238/132823374-e26be0e6-9fe6-4d37-9913-4e2859ad66c7.png)
- `logs`
master:
![image](https://user-images.githubusercontent.com/8863238/132823926-8575f6bd-fc33-4647-9323-1078a0e554e2.png)
PR:
![image](https://user-images.githubusercontent.com/8863238/132823879-1b2e7af0-8aa3-40d5-8ff9-4c61959c032b.png)
- `rm`
master:
![image](https://user-images.githubusercontent.com/8863238/132824760-1f493912-1e82-4b37-a28a-8d4bf98c8362.png)
PR:
![image](https://user-images.githubusercontent.com/8863238/132824780-6f1593d5-d313-4dac-9bdf-0ed56986ae7f.png)

The idea would be aiming to have an output as similar as possible to the original, but trying to be consistent when displaying success/error messages.

While reviewing the PR I was wondering if the `==> SUCCESS/ERROR` is too verbose for one-line outputs... The original idea behind the prefix `==>` was to delimit sections in verbose outputs such as the `reana-client validate`, where multiple lines are always displayed. But I'm not very sure if it looks nice in short outputs like:
```
==> SUCCESS: code/gendata.C was successfully moved to code/gendata.c.
```

What do you think?

Sinclert commented:
I see.

Yeah, I was assuming the [`display_message`](https://github.com/reanahub/reana-client/blob/3814b05465dc79ad682a14c84e3ae22db1b4eea8/reana_client/printer.py#L21-L59) function was perfectly figured out, so the only change missing was the refactoring of old `click.echo` and `click.secho` occurrences to the centralized custom func. **Now I see there is also work to do there.**

Regarding a _similar output_, I think we would need to consider **case by case**. For instance, in the examples you provided:

### 👉🏻  ping
I think there are perfectly fine. Having the output in green would be nice, but not _strictly_ mandatory.

### ⚙️ run
I see that messages regarding what workflow is being created (`workflow.2`), and those regarding what files are being uploaded (`File ... was successfully uploaded`), have an undesired indentation (`==>`). I agree that should be removed.

In terms of the color, again, maybe a nice to have, but no _strictly_ mandatory.

### 📑 diff
Here there is one case where I wold consider color to be **very important**. We should change the `display_message` function to also _style_ the displayed message with the same color as the padding.

### 🗑️ rm
I agree the padding is a bit too much. We _could_ leave an empty line with the green-colored `==> Success` message, having the other two without any padding; or just remove the padding of the existing two lines.

---

Unless further suggestions are given, I list the following pending changes:

- [ ] Extend `display_message` func to allow the application of color in certain messages (**relevant** in the `diff` command).
- [ ] Remove padding on the `run` command output (on the non-indented `SUCCESS: ...` lines).
- [ ] Apply color to the `diff` command output (both green and red).
- [ ] Decide whether or not we want green colored messages, upon success, on:
    - The `ping` command.
    - The `run` command.
    - The `rm` command.

---

Please comment if there is something missing.

Sinclert commented:
Hi @mvidalgarcia ,

I just rebased the branch of this PR with respect to `upstream/master`, given that there were changes producing conflicts on [cli/ping.py](https://github.com/reanahub/reana-client/blob/master/reana_client/cli/ping.py) and [utils.py](https://github.com/reanahub/reana-client/blob/master/reana_client/utils.py) modules. This was due to the recent merging of PR https://github.com/reanahub/reana-client/pull/550.

**I suggest we iterate quickly on any comments you guys have about this refactor, as it involved lots of files, and it is probable that it may raise conflicts again when other PRs are merged ahead of it.**

---

I addressed, what I would consider the most important comments of [your previous message](https://github.com/reanahub/reana-client/pull/553#pullrequestreview-751181016), on commits https://github.com/Sinclert/reana-client/commit/45ad2ad36dcb5458c7ce2bf1cc0511239777a46e and https://github.com/Sinclert/reana-client/commit/6226c52d12b616ee2cf5ba27d275281f79b418c2.

There is still the _green-colored messages_ point to be decided: is message-color that important in commands `ping`, `run` and `rm`? if so, would you rather extend [`display_message`](https://github.com/reanahub/reana-client/blob/f48a2e65950214ed530c1f454886acd512009d80/reana_client/printer.py#L21-L59) to accomodate color preference, or return to `click.secho` calls?

VMois commented:
My suggestions

Outputs from PR:

1. ping (looks good to me):

<img width="405" alt="Screenshot 2021-09-16 at 10 05 07" src="https://user-images.githubusercontent.com/15694232/133579260-22b1542c-f1ae-4ef2-b048-85ade6f3a64f.png">

2. list (looks good to me):

<img width="877" alt="Screenshot 2021-09-16 at 10 06 23" src="https://user-images.githubusercontent.com/15694232/133579438-374666d3-44bc-41f1-acea-9b512b79c27e.png">

3. create/upload/run:

<img width="930" alt="Screenshot 2021-09-16 at 10 08 16" src="https://user-images.githubusercontent.com/15694232/133579535-8b9d7981-2121-4035-9f0f-7f6a26ff75ef.png">

```
==> SUCCESS hello1.1
```
The above message looks too much to me. Would be good to use `display_message(msg_type=None)`

4. diff

<img width="1319" alt="Screenshot 2021-09-16 at 10 09 58" src="https://user-images.githubusercontent.com/15694232/133580192-f161b551-75e7-44cb-9c14-18856422c21b.png">

It is better to keep diff as it was before this PR (`click.secho`)

5. logs

<img width="1340" alt="Screenshot 2021-09-16 at 10 14 33" src="https://user-images.githubusercontent.com/15694232/133580511-30c972a6-5188-4e67-aac8-90cea5ac6593.png">

Keep as it was before PR (`click.secho`)

6. rm

<img width="762" alt="Screenshot 2021-09-16 at 10 15 15" src="https://user-images.githubusercontent.com/15694232/133580791-7d81e0b2-8f14-489c-a163-806312b77f4b.png">

Use `display_message(msg_type=None)` here

VMois commented:
Extending `display_message` at this point will prolong PR even further. In my opinion, it is fine to keep `click.secho` at some places. Later, we can gradually update if needed.

mvidalgarcia commented on reana_client/utils.py:
```diff
@@ -420,8 +396,8 @@ def run_command(cmd, display=True, return_output=False, stderr_output=False):
     if display:
-        click.secho("[{0}] ".format(now), bold=True, nl=False, fg="green")
-        click.secho("{0}".format(cmd), bold=True)
+        display_message("[{0}] ".format(now), msg_type="success")
+        display_message("{0}".format(cmd))
```
ATM we don't have any `run_command` with `display=True` but I think this can be reverted to `click.secho` and it's basically printing the date and the cmd, not a message _per se_, and we don't need SUCCESS prefix.

Sinclert replied:
Reverted to its original version

mvidalgarcia commented on reana_client/utils.py:
```diff
@@ -431,8 +407,8 @@ def run_command(cmd, display=True, return_output=False, stderr_output=False):
         if display:
-            click.secho("[{0}] ".format(now), bold=True, nl=False, fg="green")
-            click.secho("{0}".format(err), bold=True, fg="red")
+            display_message("[{0}] ".format(now), msg_type="success")
+            display_message("{0}".format(err), msg_type="error")
```
Same here.

Sinclert replied:
Reverted to its original version

mvidalgarcia reviewed (commented):
I'm leaving a bunch of comments that should cover everything and then we can merge. Thanks so much for the effort again, and I'm sorry that it got more cumbersome than expected.

> There is still the green-colored messages point to be decided: is message-color that important in commands ping, run and rm? if so, would you rather extend display_message to accomodate color preference, or return to click.secho calls?

Please keep the same colors as before, **it's OK to keep `click.secho` calls** for now.
It's a bit confusing that error messages are not printed in red as they used to. E.g. a simple `reana-client ping` call failing is white:
![image](https://user-images.githubusercontent.com/8863238/133589175-05c20160-3d5c-4e29-a87c-69332eac83f0.png)
At a first glance it looks that is fine but it isn't. I guess the same applies to other commands like `run`, `diff`, `rm`... 

Summarizing things to fix:
- `ping`: Color output of as before
- `create/upload/run`: Color output of  as before. Look at `==> SUCCESS hello1.1` message from @VMois as well.
- `logs`: still printing a lot of success/warning, it shouldn't. Color output as before.
- `rm`: doesn't look that bad as it is, no strong opinion. Either like it is or as it was before. The same happens for `download`.

Conflicts might be generated and harder to resolve due to the number of commits. Could you please squash your commits in one?

Thanks again 🙏

Sinclert commented on reana_client/cli/files.py:
```diff
@@ -318,59 +309,41 @@ def upload_files(ctx, workflow, filenames, access_token):  # noqa: D301
-                                    "uploaded.".format(file_),
-                                    fg="green",
-                                )
+                            display_message(
+                                "File {} was successfully uploaded.".format(file_)
```
Given what @VMois [commented here](https://github.com/reanahub/reana-client/pull/553#issuecomment-920711731) (`create/upload/run` section): _"This padded message is too much"_, I would rather switch to `click.secho` here.

Although it would inconsistent with the padded errors shown in the exception handling block below, **it is consistent with [your previous comment](https://github.com/reanahub/reana-client/pull/553#discussion_r709977083)** (a.k.a. _when there is a symbolic just switch to `click.secho`_).

Sinclert replied:
Changes in https://github.com/reanahub/reana-client/pull/553/commits/841d038bea862dbc40c6888640112d807fc4e4a5

Sinclert commented on reana_client/cli/utils.py:
```diff
@@ -378,25 +348,17 @@ def output_user_friendly_logs(workflow_logs, steps):
-                            mark=leading_mark,
-                            description=key_to_description_mapping[key],
+                    display_message(
+                        "{description}: {value}".format(
+                            description=key_to_description_mapping[key], value=value,
                         ),
-                        fg=status_to_color_mapping.get(logs_info["status"]),
+                        msg_type=JOB_STATUS_TO_MSG_TYPE.get(logs_info["status"]),
                     )
```
m...m... not sure what _same here_ means in this context.

In its original version, this code snippet also considered the _job_status_ to display the message in different colors. If you want to preserve the colors, and avoid the paddings (`==>`), I would need to revert to its original version (using click.secho`).

mvidalgarcia replied:
Exactly, that's what I meant (revert back). Sorry if I wasn't explicit enough. We want to avoid all these `SUCCESS/ERROR` messages:
![image](https://user-images.githubusercontent.com/8863238/133633644-e89d28c6-2149-43cf-ad31-ce47ce35ac90.png)

Sinclert replied:
Changed in https://github.com/reanahub/reana-client/commit/49de64be82dd4c968ef8edd44d71434c39d06053

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -1225,41 +1199,30 @@ def print_color_diff(lines):
-                    "{} No differences in REANA specifications.".format(leading_mark),
-                    bold=True,
-                    fg="yellow",
+                display_message(
+                    "No differences in REANA specifications.", msg_type="warning"
```
Set as `info` messages, as they are used as separators.

mvidalgarcia replied:
It would be nice to keep them yellow though.

Sinclert replied:
Changed in https://github.com/reanahub/reana-client/pull/553/commits/b24c7df30008334afd328f47be3c1fdfbfb67a13

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -1225,41 +1199,30 @@ def print_color_diff(lines):
-                    "{} Differences in workflow {}".format(leading_mark, section),
-                    bold=True,
-                    fg="yellow",
+                display_message(
+                    "Differences in workflow {}".format(section), msg_type="warning"
```
Set as `info` messages, as they are used as separators.

Sinclert replied:
Changed in https://github.com/reanahub/reana-client/pull/553/commits/b24c7df30008334afd328f47be3c1fdfbfb67a13

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -1225,41 +1199,30 @@ def print_color_diff(lines):
-                bold=True,
-                fg="yellow",
-            )
+            display_message("Differences in workflow workspace", msg_type="warning")
```
Set as `info` messages, as they are used as separators.

mvidalgarcia replied:
t would be nice to keep them yellow though.

Sinclert replied:
Changed in https://github.com/reanahub/reana-client/pull/553/commits/b24c7df30008334afd328f47be3c1fdfbfb67a13

Sinclert commented on reana_client/cli/files.py:
```diff
@@ -318,59 +309,41 @@ def upload_files(ctx, workflow, filenames, access_token):  # noqa: D301
-                                )
+                            display_message(
+                                "Symlink resolved to {}. "
+                                "Uploaded hard copy.".format(file_[len("symlink:") :])
```
Changes in https://github.com/reanahub/reana-client/pull/553/commits/841d038bea862dbc40c6888640112d807fc4e4a5

Sinclert commented on reana_client/cli/ping.py:
```diff
@@ -39,34 +40,30 @@ def ping(ctx, access_token):  # noqa: D301
-                ),
-                fg=msg_color,
-            )
+        display_message(
```
Changes in https://github.com/reanahub/reana-client/pull/553/commits/115fea778678a4fe7eb8ddbf47327a0065a39485

Sinclert commented on reana_client/cli/utils.py:
```diff
@@ -330,19 +306,16 @@ def output_user_friendly_logs(workflow_logs, steps):
-            "{} Workflow engine logs".format(leading_mark), bold=True, fg="yellow"
-        )
-        click.echo(workflow_logs["workflow_logs"])
+        display_message("Workflow engine logs", msg_type="warning")
```
Changes in https://github.com/reanahub/reana-client/pull/553/commits/49de64be82dd4c968ef8edd44d71434c39d06053

Sinclert commented on reana_client/cli/utils.py:
```diff
@@ -330,19 +306,16 @@ def output_user_friendly_logs(workflow_logs, steps):
-        click.secho("{} Engine internal logs".format(leading_mark), fg="yellow")
-        click.secho(workflow_logs["engine_specific"])
+        display_message("\n")
+        display_message("Engine internal logs", msg_type="warning")
```
Changes in https://github.com/reanahub/reana-client/pull/553/commits/49de64be82dd4c968ef8edd44d71434c39d06053

Sinclert commented on reana_client/cli/utils.py:
```diff
@@ -351,25 +324,22 @@ def output_user_friendly_logs(workflow_logs, steps):
-        click.echo("\n")
-        click.secho("{} Job logs".format(leading_mark), bold=True, fg="yellow")
+        display_message("\n")
+        display_message("Job logs", msg_type="warning")
```
Changes in https://github.com/reanahub/reana-client/pull/553/commits/49de64be82dd4c968ef8edd44d71434c39d06053

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -1313,27 +1276,26 @@ def workflow_open_interactive_session(
                 interactive_session_configuration,
             )
-            click.secho(
+            display_message(
```
Changed in https://github.com/reanahub/reana-client/pull/553/commits/504fe0719776b84434e94d880145d35a88493cae

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -1357,17 +1319,16 @@ def workflow_close_interactive_session(workflow, access_token):  # noqa: D301
             logging.info("Closing an interactive session on {}".format(workflow))
             close_interactive_session(workflow, access_token)
-            click.echo(
+            display_message(
```
Changed in https://github.com/reanahub/reana-client/pull/553/commits/2a60bc716559505fb432d46701c954a14da28e70

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -333,7 +331,7 @@ def workflow_create(ctx, file, name, skip_validation, access_token):  # noqa: D3
         logging.info("Connecting to {0}".format(get_api_url()))
         response = create_workflow(reana_specification, name, access_token)
-        click.echo(click.style(response["workflow_name"], fg="green"))
+        display_message(response["workflow_name"], msg_type="success")
```
Changed in https://github.com/Sinclert/reana-client/commit/668e8e7f44afebdfe3fc7535ad4ddc93588af35a

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -421,36 +419,31 @@ def workflow_start(
-            click.secho(
-                get_workflow_status_change_msg(workflow, current_status), fg="green"
-            )
+            display_message(get_workflow_status_change_msg(workflow, current_status))
```
Changed in https://github.com/Sinclert/reana-client/commit/668e8e7f44afebdfe3fc7535ad4ddc93588af35a

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -421,36 +419,31 @@ def workflow_start(
-                        get_workflow_status_change_msg(workflow, current_status),
-                        fg="green",
+                    display_message(
+                        get_workflow_status_change_msg(workflow, current_status)
```
Changed in https://github.com/Sinclert/reana-client/commit/668e8e7f44afebdfe3fc7535ad4ddc93588af35a

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -421,36 +419,31 @@ def workflow_start(
-                            click.secho(
-                                "[INFO] Listing workflow output " "files...", bold=True
-                            )
+                            display_message("Listing workflow output files...")
```
Changed in https://github.com/Sinclert/reana-client/commit/668e8e7f44afebdfe3fc7535ad4ddc93588af35a

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -806,14 +792,12 @@ def workflow_logs(
-                                ),
-                                fg="red",
+                        display_message(
+                            "Error: filter '{}' is not valid.\n"
```
Changed in https://github.com/Sinclert/reana-client/commit/49de64be82dd4c968ef8edd44d71434c39d06053

Sinclert commented on reana_client/cli/workflow.py:
```diff
@@ -826,26 +810,22 @@ def workflow_logs(
-                        ),
-                        fg="red",
+                display_message(
+                    "Error: please provide complete --filter name=value pairs, "
```
Changed in https://github.com/Sinclert/reana-client/commit/49de64be82dd4c968ef8edd44d71434c39d06053

Sinclert commented:
Hi @mvidalgarcia and @VMois ,

As you can see I addressed every single comment you had, linking, under your comments, the commits in which I addressed them. This is the reason I have no _squashed_ the commits yet. I think it is easier to review this way 😄 

As a last comment, please noticed I had to correct a hard comparison within one of the tests (check https://github.com/Sinclert/reana-client/commit/d11ea3fb79b153f05cfc07947c09a49cf58d5832).

---

**Please, review the changes. If everything is good to go I will go ahead and squash before merging**

mvidalgarcia commented:
Hi @Sinclert,

I went ahead and created a commit with the changes I think we need, hope you don't mind. Otherwise, I'm afraid we're going to be interacting back a forth for a while. Please check that everything looks fine, if so, proceed with the squashing.

I know it wasn't pleasant to deal with this task and we should learn from it. Maybe next time we can first create an issue detailing what's needed and have some prior discussion there before creating a PR.

Thanks so much for your contribution and sorry for the misunderstandings.

Sinclert commented:
I do not mind at all :)

At the end of the day, it is your choice, as maintainer, the one to preserve.

Regarding the painful revision process, I agree, I should have commented on its associated issue (https://github.com/reanahub/reana-client/issues/431) before jumping into code changes. Just as a reflection point (not an excuse), I believe everything started to get out of hand when the _original_ purpose of this PR (direct simplification of `click.echo(click.style(...)))` by `click.secho(...)`) switched to _let's use [`display_message`](https://github.com/reanahub/reana-client/blob/f48a2e65950214ed530c1f454886acd512009d80/reana_client/printer.py#L21-L59) throughout the codebase to unify how -certain- code messages are shown_.

I would also improve my handling of these situations for next time.
---

Proceeding to squash...

Sinclert commented:
Just squashed everything in a handful of commits. I have preserved some of them that are different in some aspect to the original two (https://github.com/reanahub/reana-client/pull/553/commits/3f98f5824ab01c1db3ba96710983e0ba74713129 and https://github.com/reanahub/reana-client/pull/553/commits/7f8358ca65a2a0e58b6089d65f5fd618e48540b4):

- Commit https://github.com/reanahub/reana-client/pull/553/commits/fc47e37654ddab4d3eaca9d6ca78485021f99ce2 has a different scope: it just cleans up unused imports.
- Commit https://github.com/reanahub/reana-client/pull/553/commits/c20cb4d895282ce3a004430dba907a5d12bcfd74 reverts some of the changes, but I preserved it as it has **all** your suggested changes from [this comment](https://github.com/reanahub/reana-client/pull/553#issuecomment-920370199) forward.
- Commit https://github.com/reanahub/reana-client/pull/553/commits/22dc668c93eca010b7816d0b97fc5d4c1838e6f8 just affect a different part of the codebase: tests.
- Commit https://github.com/reanahub/reana-client/pull/553/commits/9eec129d7ca64ba8e16cb3e752756a302d4464da was signed by @mvidalgarcia , and I think it is nice to keep it separated.

mvidalgarcia reviewed (approved):
_Merging..._
